test/lib/ufe/testRename.py

```diff
@@ -149,7 +166,52 @@ def childrenNames(children):
         propsChildren = propsHierarchy.children()
         propsChildrenNames = childrenNames(propsChildren)
 
-        self.assertIn(ball35RenName, propsChildrenNames)
-        self.assertNotIn('Ball_35', propsChildrenNames)
+        self.assertIn(pCylinder1RenName, propsChildrenNames)
+        self.assertNotIn('pCylinder1', propsChildrenNames)
         self.assertEqual(len(propsChildren), len(propsChildrenPre))
 
+    def testRenameRestriction(self):
+        '''Restrict renaming USD node. Cannot rename a prim defined on another layer.'''
+
+        # select a USD object.
+        mayaPathSegment = mayaUtils.createUfePathSegment('|world|transform1|proxyShape1')
+        usdPathSegment = usdUtils.createUfePathSegment('/Room_set/Props/Ball_35')
+        ball35Path = ufe.Path([mayaPathSegment, usdPathSegment])
+        ball35Item = ufe.Hierarchy.createItem(ball35Path)
+
+        ufe.GlobalSelection.get().append(ball35Item)
+
+        # get the USD stage
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+
+        # check GetLayerStack behavior
+        self.assertEqual(stage.GetLayerStack()[0], stage.GetSessionLayer())
+        self.assertEqual(stage.GetEditTarget().GetLayer(), stage.GetSessionLayer())
+
+        # expect the exception happens
+        with self.assertRaises(RuntimeError):
+            newName = 'Ball_35_Renamed'
+            cmds.rename(newName)
+
+    def testRenameRestriction2(self):
+        '''Restrict renaming USD node. Cannot rename a prim with definitions or opinions on other layers.'''
+
+        # select a USD object.
+        mayaPathSegment = mayaUtils.createUfePathSegment('|world|transform1|proxyShape1')
+        usdPathSegment = usdUtils.createUfePathSegment('/Room_set/Props/Ball_35')
+        ball35Path = ufe.Path([mayaPathSegment, usdPathSegment])
+        ball35Item = ufe.Hierarchy.createItem(ball35Path)
+
+        ufe.GlobalSelection.get().append(ball35Item)
+
+        # get the USD stage
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+
+        # set the edit target to Assembly_room_set.usda
+        stage.SetEditTarget(stage.GetLayerStack()[2])
+        self.assertEqual(stage.GetEditTarget().GetLayer().GetDisplayName(), "Assembly_room_set.usda")
+
+        # expect the exception happens
+        with self.assertRaises(RuntimeError):
+           newName = 'Ball_35_Renamed'
+           cmds.rename(newName)
```

Comment:
@ppt-adsk  I added two test cases for rename restriction operations. I created two separate tests:

`testRenameRestriction` is to test if the prim that we are about to rename is in the correct layer

`testRenameRestriction2` is to test if the prim that we are about to rename doesn't have any definitions or opinions on other layers.

If you think `testRenameRestriction2` is ugly, I am happy to change it with something more meaningful.